cpp/src/arrow/compute/kernels/scalar_compare_test.cc

```diff
@@ -1310,6 +1348,217 @@ TYPED_TEST(TestVarArgsCompareParametricTemporal, MinElementWise) {
                {this->array("[1, null, 3, 4]"), this->array("[2, 2, null, 2]")});
 }
 
+TYPED_TEST(TestVarArgsCompareBinary, MinElementWise) {
```

Comment:
These are very comprehensive tests, good...although some of them may be a bit redundant, but ok.
What happens if input consists of multiple arrays of different lengths?

Reply:
I have a test case with different lengths right at the bottom of `TestVarArgsCompareBinary` suites. I agree that it deserves several more, my plan is to take care of completing the tests tomorrow. The decimal test suite is insufficient too (just one test case).

These tests are copied and adjusted from the numeric ones, I haven't tried to prune them yet. They do a decent job at exercising various combinations of null, empty and scalar / array mixing, but I agree we could achieve similar coverage with less and add more different length coverage.

Reply:
Done, I've added more extensive tests for strings and decimals.